editor/plugins/polygon_2d_editor_plugin.cpp

```diff
@@ -52,38 +53,37 @@
 #include "scene/gui/texture_rect.h"
 #include "scene/gui/view_panner.h"
 
-class UVEditDialog : public AcceptDialog {
-	GDCLASS(UVEditDialog, AcceptDialog);
-
-	void shortcut_input(const Ref<InputEvent> &p_event) override {
-		const Ref<InputEventKey> k = p_event;
-		if (k.is_valid() && k->is_pressed()) {
-			bool handled = false;
-
-			if (ED_IS_SHORTCUT("ui_undo", p_event)) {
-				EditorNode::get_singleton()->undo();
-				handled = true;
-			}
-
-			if (ED_IS_SHORTCUT("ui_redo", p_event)) {
-				EditorNode::get_singleton()->redo();
-				handled = true;
-			}
-
-			if (handled) {
-				set_input_as_handled();
-			}
-		}
-	}
-};
-
 Node2D *Polygon2DEditor::_get_node() const {
 	return node;
 }
 
 void Polygon2DEditor::_set_node(Node *p_polygon) {
+	CanvasItem *draw = Object::cast_to<CanvasItem>(canvas);
+	if (node) {
+		node->disconnect("draw", callable_mp(draw, &CanvasItem::queue_redraw));
+		node->disconnect("draw", callable_mp(this, &Polygon2DEditor::_update_available_modes));
+	}
+
 	node = Object::cast_to<Polygon2D>(p_polygon);
 	_update_polygon_editing_state();
+	canvas->queue_redraw();
+	if (node) {
+		canvas->set_texture_filter(node->get_texture_filter_in_tree());
+
+		if (!polygon_edit->is_inside_tree()) {
+			EditorNode::get_bottom_panel()->add_item(TTR("Polygon"), polygon_edit);
```

Comment:
That's not how bottom panel is usually used. You should add it at the beginning and only hide/show the button. Also it's missing shortcut like other editors. See `editor\plugins\tiles\tiles_editor_plugin.cpp` for example.